src/v/cluster/archival_metadata_stm.cc

```diff
@@ -267,6 +267,18 @@ ss::future<stm_snapshot> archival_metadata_stm::take_snapshot() {
 }
 
 model::offset archival_metadata_stm::max_collectible_offset() {
+    if (
+      _raft->log_config().has_overrides()
+      && !model::is_archival_enabled(
+        _raft->log_config().get_overrides().shadow_indexing_mode)) {
+        // The archival is disabled but the state machine still exists so we
+        // shouldn't stop eviction from happening.
+        vlog(
+          _logger.info,
+          "handled log eviction, next offset: {}",
```

Comment:
That would probably lead to excessive logging. Also log message seems a bit misleading, maybe delete it?